File: backend/OlivePlatform/OlivePlatform.Application/Features/AppConstants/Responses/AppConstantsResponse.cs
namespace OlivePlatform.Application.Features.AppConstants.Responses
{
    public class AppConstantsResponse
    {

        public IReadOnlyList<AppConstantItemResponse> OliveVarieties { get; init; } = [];

        public IReadOnlyList<AppConstantItemResponse> PurchaseStatus { get; init; } = [];

        public IReadOnlyList<AppConstantItemResponse> ProductionStatus { get; init; } = [];

        public IReadOnlyList<AppConstantItemResponse> OilMovementTypes { get; init; } = [];

        public IReadOnlyList<AppConstantItemResponse> InvoiceTypes { get; init; } = [];

        public IReadOnlyList<AppConstantItemResponse> InvoiceStatuses { get; init; } = [];

        public IReadOnlyList<AppConstantItemResponse> PaymentMethods { get; init; } = [];

    }
}

File: frontend/olive-platform/src/features/appConstants/data/responses/AppConstantsResponse.ts
import type { AppConstantItemResponse } from "./AppConstantItemResponse";

export type AppConstantsResponse = {
  oliveVarieties: AppConstantItemResponse[];
  productionStatus: AppConstantItemResponse[];
  oilMovementTypes: AppConstantItemResponse[];
  invoiceTypes: AppConstantItemResponse[];
  invoiceStatuses: AppConstantItemResponse[];
  paymentMethods: AppConstantItemResponse[];
  purchaseStatus: AppConstantItemResponse[];
};
